packages/monaco/src/browser/monaco-frontend-module.ts

```diff
@@ -138,18 +136,14 @@ export function createMonacoConfigurationService(container: interfaces.Container
     const service = monaco.services.StaticServices.configurationService.get();
     const _configuration = service._configuration;
 
-    const getValue = _configuration.getValue.bind(_configuration);
     _configuration.getValue = (section, overrides, workspace) => {
-        const preferenceConfig = configurations.getConfiguration();
+        const overrideIdentifier = overrides && 'overrideIdentifier' in overrides && overrides['overrideIdentifier'] as string || undefined;
+        const resourceUri = overrides && 'resource' in overrides && overrides['resource'].toString();
+        const proxy = preferences.getPreferences(resourceUri, overrideIdentifier);
         if (section) {
-            const value = preferenceConfig.get(section);
-            return value !== undefined ? value : getValue(section, overrides, workspace);
-        }
-        const simpleConfig = getValue(section, overrides, workspace);
-        if (typeof simpleConfig === 'object') {
-            return deepmerge([{}, simpleConfig, preferenceConfig.toJSON()]);
+            return proxy[section];
```

Comment:
strange that it works, if a section is `foor.bar` then a configuration (i.e. each subsection should be an object) should be:
```json
{
  foo: {
    bar: ...
  }
}
```

It seems to be:
```json
{
  foo.bar: ...
}
```

Reply:
it has both

Reply:
then we will send duplicate data to language servers, i've tried to add test: https://github.com/eclipse-theia/theia/pull/6172#discussion_r324964082

but it fails with some serialization errors

Reply:
fixed